app/pages/patient/DataDonationConsentDialog.js

```diff
@@ -0,0 +1,362 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { withTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { Box, Flex, Text } from 'theme-ui';
+import * as yup from 'yup';
+import { getCommonFormikFieldProps, fieldsAreValid } from '../../core/forms';
+import { useFormik } from 'formik';
+import { isString, keys, memoize } from 'lodash';
+import Markdown from 'react-markdown';
+import InfoRoundedIcon from '@material-ui/icons/InfoRounded';
+
+import Button from '../../components/elements/Button';
+import Checkbox from '../../components/elements/Checkbox';
+import TextInput from '../../components/elements/TextInput';
+import { Body0, MediumTitle, Paragraph0, Paragraph1, Paragraph2, Subheading } from '../../components/elements/FontStyles';
+
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+} from '../../components/elements/Dialog';
+
+import i18next from '../../core/language';
+import { colors, shadows } from '../../themes/baseTheme';
+import personUtils from '../../core/personutils';
+import Pill from '../../components/elements/Pill';
+
+import { DATA_DONATION_CONSENT_TYPE } from '../../core/constants';
+
+const t = i18next.t.bind(i18next);
+
+export const getConsentText = memoize((accountType, patientAgeGroup, patientName, caregiverName, consentDate) => {
```

Comment:
This is the main method to create the various bits of text content based on account type and patient age.  I memoized it since it does get called from a few places, and the arguments rarely change.